### PR TITLE
Add async native support and support to async_trait

### LINKED ISSUE #4: The #[time] macro seems to strip the "unsafe" keyword from a decorated function
If I write an unsafe function and apply the `#[time]` decorator macro:

```
#[time]
pub unsafe fn do_something_risky() { ... }
```

I am now able to call this function outside of an unsafe block, e.g.

```
fn main() {
    do_something_risky();
}
```

Whereas I would expect to have to write:

```
fn main() {
    unsafe { 
        do_something_risky();
    }
}
```

If `do_something_risky()` needs to be `unsafe` to call other unsafe functions, then fortunately this generates a compiler error, but for other unsafe behaviour that the compiler can't know about, for example multi-threading invariants documented by the programmer, this is pretty serious, because there's currently no way for the programmer to _require_ `unsafe` when calling such a decorated function.

A workaround is to create an unsafe wrapper function, so that `#[time]` is only on the inner function, but this is quite messy.

### PR DESCRIPTION
**Pull request by S0c5**
Small change to include asyncness , it closes #4

### CODE REVIEW

**Comment by PhilipDaniels** [comment]
David, thanks for the PR, especially updating the example. There is actually an open issue #3 which this will hopefully fix. You can probably tell from my Github history that I have been away from the computer for a while, but I will try and look at this in detail later this week.

Two comments based on my initial observations

1. For an async fn, the time reported is going to be the total wall clock time, which may be far larger than the time the fn spends executing on the CPU. While this is true for any fn (due to pre-emption), for async fns it is particularly notable because they are far more likely to be suspended. I think this should be mentioned in the documentation to avoid confusion.
2. Yesterday (your PR is very good timing) another user reported in #4 that the `unsafe` qualifier gets stripped - pretty much in the same way that `async` gets stripped. There is an `unsafety` field in https://docs.rs/syn/latest/syn/struct.Signature.html. It would be very much appreciated if you could also capture that in this PR and therefore fix both bugs at the same time!

File: rust-toolchain [hunk 1]
```diff
@@ -0,0 +1 @@
+nightly
```
Comment (PhilipDaniels) [inline rust-toolchain]: Why did you add this?

**Comment by S0c5** [comment]
@PhilipDaniels amazing, I just updated the lib to add support for async_trait.

**Comment by PhilipDaniels** [comment]
Thanks for the updates, I'll take a look on Sunday and release it then if all is ok.

**Comment by PhilipDaniels** [comment]
v1.1.1 published just now.
